sdk/internal/mock/mock.go

```diff
@@ -123,7 +127,20 @@ func (s *Server) Do(req *http.Request) (*http.Response, error) {
 		resp := s.getResponse()
 		return nil, resp.err
 	}
-	resp, err := s.srv.Client().Do(req)
+	var err error
+	var resp *http.Response
+	if s.routeAllRequestsToMockServer {
+		originalURL := req.URL
+		mockUrl := *req.URL
+		srvUrl, _ := url.Parse(s.srv.URL)
```

Comment:
Unlikely, but if there is an error here we should return the error and an empty `&http.Response{}`

Reply:
I'm not sure it is worth handling this extreme edge case.

Reply:
Handling unlikely errors is the quintessence of Go programming 🤣 

But shouldn't we return `nil` instead of `&http.Response{}`?

Reply:
Fair enough 😄